emscripten.py

```diff
@@ -1313,6 +1313,14 @@ def emscript_wasm_backend(infile, settings, outfile, libraries=None, compiler_en
     backend_args += ['-thread-model=single'] # no threads support in backend, tell llc to not emit atomics
     # disable slow and relatively unimportant optimization passes
     backend_args += ['-combiner-alias-analysis=false', '-combiner-global-alias-analysis=false']
+
+    # asm.js-style exception handling
+    if settings['DISABLE_EXCEPTION_CATCHING'] != 1:
+      backend_args += ['-enable-emscripten-cxx-exceptions']
+    if settings['DISABLE_EXCEPTION_CATCHING'] == 2:
```

Comment:
Actually in the asm2wasm path, it is safe to do `fake` because of the prefix (a compiled method would start with `_`). So this is another case where the wasm backend not prefixing causes a difference. We should either change the asm.js path to use `___fake` (triple underscore, so it's really the double, otherwise it's just the name of a single-prefixed function, which by the C standard means we could collide, I believe?), or leave it `fake` as it is. Seems simplest to leave it.